BGL/include/CGAL/boost/graph/named_params_helper.h

```diff
@@ -112,6 +114,204 @@ namespace CGAL {
     typedef typename boost::property_traits<PMap>::value_type type;
   };
 
+  namespace internal_np
+  {
+  template<typename key_value, class PMap, class Graph, typename Tag>
+  struct Index_map_initializer{
+    void operator()(PMap, const Graph& )
+    {}
+  };
+
+  template< class PMap, class Graph>
+  struct Index_map_initializer<
+      typename boost::graph_traits<Graph>::vertex_descriptor,
+      PMap, Graph,
+      CGAL::Tag_true>{
+    void operator()(PMap map, const Graph& g)
+    {
+      CGAL::helpers::init_vertex_indices(g, map);
+    }
+  };
+
+  template< class PMap, class Graph>
+  struct Index_map_initializer<
+      typename boost::graph_traits<Graph>::halfedge_descriptor,
+      PMap, Graph,
+      CGAL::Tag_true>{
+    void operator()(PMap map, const Graph& g)
+    {
+      CGAL::helpers::init_halfedge_indices(g, map);
+    }
+  };
+
+  template< class PMap, class Graph>
+  struct Index_map_initializer<
+      typename boost::graph_traits<Graph>::face_descriptor,
+      PMap, Graph,
+      CGAL::Tag_true>{
```

Comment:
Any code design that supposes that `vertex_descriptor`, `halfedge_descriptor`, and `face_descriptor` are distinct types is broken by design. I have already said so multiple times, and we should stop assuming that the three types are distinct!

You should:
  - use the `Default_tag` or `Dynamic_tag` to distinguish between the three cases,
  - and use free functions overloads, instead of template specialization.

Reply:
free functions overloads that would decide vertex/halfedge/face initialization based on the named_parameter they receive, right ?